core/src/main/java/io/github/tcdl/adapters/MockAdapter.java

```diff
@@ -1,60 +1,73 @@
-    }
-
-    @Override
-    public void publish(String jsonMessage) throws ChannelException {
-        log.info("Sending request {}", jsonMessage);
-        requests.offer(jsonMessage);
-        log.info("Requests to process {}", requests);
-    }
-
-    @Override
-    public void subscribe(RawMessageHandler messageHandler) {
-        log.info("Subscribed. Process first message from list {}", requests);
-        this.messageHandler = messageHandler;
-        if (messageHandler != null && !requests.isEmpty()) {
-            String jsonMessage = requests.peek();
-            handleMessage(jsonMessage);
-        }
-    }
-
-    @Override
-    public void unsubscribe() {
-        messageHandler = null;
-    }
-
-    private void handleMessage(String jsonMessage) {
-        log.debug("Retrieved response {}", jsonMessage);
-        messageHandler.onMessage(jsonMessage);
-    }
-    
-    public void clearAllMessages () {
-        requests.clear();
-    }
-}
+package io.github.tcdl.adapters;
+
+import io.github.tcdl.exception.ChannelException;
+import io.github.tcdl.exception.JsonConversionException;
+import io.github.tcdl.messages.Message;
+import io.github.tcdl.support.Utils;
+
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by rdro on 4/24/2015.
+ */
+public class MockAdapter implements Adapter {
+
+    public static Logger log = LoggerFactory.getLogger(MockAdapter.class);
+
+    private static MockAdapter instance = new MockAdapter();
+
+    private Map<String, Queue<String>> messageMap = new ConcurrentHashMap<String, Queue<String>>();
+
+    private MockAdapter() {
+    }
+
+    public static MockAdapter getInstance() {
+        return instance;
+    }
+
+    @Override
+    public void publish(String jsonMessage) throws ChannelException {
+        log.info("Recieved request {}", jsonMessage);
+        try {
+            Message incommingMessage = Utils.fromJson(jsonMessage, Message.class);
+            String topicTo = incommingMessage.getTopics().getTo();
+            Queue<String> messagesQueue = messageMap.get(topicTo);
+            if (messagesQueue == null) {
+                messagesQueue = new ConcurrentLinkedQueue<String>();
+                Queue<String> curQ = messageMap.putIfAbsent(topicTo, messagesQueue);
+                if (curQ != null) {
+                    messagesQueue = curQ;
+                }
+            }
+            messagesQueue.add(jsonMessage);
+            log.info("Message for topic {} published: [{}]", topicTo, jsonMessage);
+        } catch (JsonConversionException e) {
+            log.error("Recieved message can not be parsed");
+        }
+    }
+
+    @Override
+    public void subscribe(RawMessageHandler messageHandler) {
+    }
+
+    public void subscribeForTopic(RawMessageHandler messageHandler, String topic) {

```

Comment:
Maybe it would be better to create separate adapter per topic as implemented for AmqpAdapter and properly implement subscribe method instead of adding this public method?